pygeo/constraints/thicknessConstraint.py

```diff
@@ -88,6 +88,115 @@ def writeTecplot(self, handle):
+
+        self.coords = coords
+        self.scaled = scaled
+
+        # First thing we can do is embed the coordinates into DVGeo
+        # with the name provided:
+        self.DVGeo.addPointSet(self.coords, self.name, compNames=compNames)
+
+        # Now get the reference lengths and directions
+        self.D0 = np.zeros(self.nCon)
+        self.dir_vec = np.zeros((self.nCon, 3))
+        for i in range(self.nCon):
+            vec = self.coords[2 * i] - self.coords[2 * i + 1]
+            self.D0[i] = geo_utils.norm.euclideanNorm(vec)
+            self.dir_vec[i] = vec / self.D0[i]
+
+    def evalFunctions(self, funcs, config):
+        """
+        Evaluate the functions this object has and place in the funcs dictionary
+
+        Parameters
+        ----------
+        funcs : dict
+            Dictionary to place function values
+        """
+        # Pull out the most recent set of coordinates:
+        self.coords = self.DVGeo.update(self.name, config=config)
+        D = np.zeros(self.nCon)
+        for i in range(self.nCon):
+            vec = self.coords[2 * i] - self.coords[2 * i + 1]
+
+            # take the dot product with the direction vector
+            D[i] = vec[0] * self.dir_vec[i, 0] + vec[1] * self.dir_vec[i, 1] + vec[2] * self.dir_vec[i, 2]
+
+            if self.scaled:
+                D[i] /= self.D0[i]
+
+        funcs[self.name] = D
+
+    def evalFunctionsSens(self, funcsSens, config):
+        """
+        Evaluate the sensitivity of the functions this object has and
+        place in the funcsSens dictionary
+
+        Parameters
+        ----------
+        funcsSens : dict
+            Dictionary to place function values
+        """
+
+        nDV = self.DVGeo.getNDV()
+        if nDV > 0:
+            dTdPt = np.zeros((self.nCon, self.coords.shape[0], self.coords.shape[1]))
+            for i in range(self.nCon):
+                D_b = 1.0
+
+                if self.scaled:
+                    D_b /= self.D0[i]
+
+                # d(dot(vec,n)/dvec = n
+                vec_b = self.dir_vec[i] * D_b
+
+                dTdPt[i, 2 * i, :] = vec_b
+                dTdPt[i, 2 * i + 1, :] = -vec_b
+
+            funcsSens[self.name] = self.DVGeo.totalSensitivity(dTdPt, self.name, config=config)
+
+    def writeTecplot(self, handle):
+        """
+        Write the visualization of this set of thickness constraints
+        to the open file handle
+        """
+
+        handle.write("Zone T=%s\n" % self.name)
+        handle.write("Nodes = %d, Elements = %d ZONETYPE=FELINESEG\n" % (len(self.coords), len(self.coords) // 2))
+        handle.write("DATAPACKING=POINT\n")
+        for i in range(len(self.coords)):
+            handle.write(f"{self.coords[i, 0]:f} {self.coords[i, 1]:f} {self.coords[i, 2]:f}\n")
+
+        for i in range(len(self.coords) // 2):
+            handle.write("%d %d\n" % (2 * i + 1, 2 * i + 2))
+
+        handle.write("Zone T=%s_ref_directions\n" % self.name)
```

Comment:
I would add a comment here to explain that the projected toothpicks are saved in a different zone for visualization purposes

Reply:
Added!